src/mr_allocator.rs

```diff
@@ -147,38 +220,58 @@ impl MrAllocator {
+                    alloc_from_je(self.default_arena_ind, layout, flag).map_or_else(||{
+                            Err(io::Error::new(io::ErrorKind::OutOfMemory, "insufficient contiguous memory was available to service the allocation request"))
+                        }, |addr|{
+                            let raw_mr = self.lookup_raw_mr(addr as usize);
+                            Ok(LocalMrInner::new(addr as usize, *layout, raw_mr, self.strategy))
+                        })
+                }else {
+                    let arena_id = ACCESS_ARENA_MAP.lock().get(&access).copied();
+                    let arena_id = arena_id.map_or_else(||{
+                        // didn't use this access before, so create an arena to mange this kind of MR
+                        let ind = init_je_statics(Arc::<ProtectionDomain>::clone(&self.pd), access)?;
+                        if ACCESS_ARENA_MAP.lock().insert(access, ind).is_some(){
+                            return Err(io::Error::new(
+                                io::ErrorKind::Other,
+                                format!("this is a bug: insert ACCESS_ARENA_MAP failed, access:{:?} has been recorded", access),
+                            ))
+                        };
+                        Ok(ind)
+                    }, |ind|{Ok(ind)})?;
+
+                    alloc_from_je(arena_id, layout, flag).map_or_else(||{
+                        Err(io::Error::new(io::ErrorKind::OutOfMemory, "insufficient contiguous memory was available to service the allocation request"))
+                    }, |addr|{
+                        #[allow(clippy::unreachable)]
+                        let raw_mr = lookup_raw_mr(arena_id, addr as usize).map_or_else(
+                            || {
+                                unreachable!("can not find raw mr with arena_id: {} by addr: {}",arena_id, addr as usize);
+                            },
+                            |raw_mr| raw_mr,
+                        );
+                        Ok(LocalMrInner::new(addr as usize, *layout, raw_mr, self.strategy))
+                    })
+                }
+            },
+            MRManageStrategy::Raw => {
+                alloc_raw_mem(layout, flag).map_or_else(||{
+                        Err(io::Error::new(io::ErrorKind::OutOfMemory, "insufficient contiguous memory was available to service the allocation request"))
+                    }, |addr|{
+                        let raw_mr = Arc::new(RawMemoryRegion::register_from_pd(&self.pd, addr, layout.size(), access)?);
+                        Ok(LocalMrInner::new(addr as usize, *layout, raw_mr, self.strategy))
+                    })
+            },
         }
     }
 
     /// Look up `raw_mr` info by addr
     #[allow(clippy::unreachable)]
     fn lookup_raw_mr(&self, addr: usize) -> Arc<RawMemoryRegion> {
-        lookup_raw_mr(self.arena_ind, addr).map_or_else(
+        lookup_raw_mr(self.default_arena_ind, addr).map_or_else(
```

Comment:
what if the addr is not in the default arena?

Reply:
This fn is a wrapper for lookup default mr, maybe it is unnecessary for now, I will remove it to avoid doubt.